includes/class-wc-payments-apple-pay-registration.php

```diff
@@ -0,0 +1,357 @@
+	 */
+	private $apple_pay_verify_notice;
+
+	/**
+	 * Initialize class actions.
+	 *
+	 * @param WC_Payments_API_Client $payments_api_client WooCommerce Payments API client.
+	 */
+	public function __construct( WC_Payments_API_Client $payments_api_client ) {
+		add_action( 'init', [ $this, 'add_domain_association_rewrite_rule' ] );
+		add_action( 'admin_init', [ $this, 'verify_domain_on_domain_name_change' ] );
+		add_action( 'admin_notices', [ $this, 'admin_notices' ] );
+		add_filter( 'query_vars', [ $this, 'whitelist_domain_association_query_param' ], 10, 1 );
+		add_action( 'parse_request', [ $this, 'parse_domain_association_request' ], 10, 1 );
+
+		add_action( 'woocommerce_woocommerce_payments_updated', [ $this, 'verify_domain_if_configured' ] );
+		add_action( 'add_option_woocommerce_woocommerce_payments_settings', [ $this, 'verify_domain_on_new_settings' ], 10, 2 );
+		add_action( 'update_option_woocommerce_woocommerce_payments_settings', [ $this, 'verify_domain_on_updated_settings' ], 10, 2 );
+
+		// Add default settings in case the payment request options are missing.
+		$this->gateway_settings        = array_merge( WC_Payments_Payment_Request::get_default_settings(), get_option( 'woocommerce_woocommerce_payments_settings', [] ) );
+		$this->domain_name             = isset( $_SERVER['HTTP_HOST'] ) ? $_SERVER['HTTP_HOST'] : str_replace( [ 'https://', 'http://' ], '', get_site_url() ); // @codingStandardsIgnoreLine
+		$this->apple_pay_verify_notice = '';
+		$this->payments_api_client     = $payments_api_client;
+	}
+
+	/**
+	 * Gets the Gateway settings.
+	 *
+	 * @param string $setting Setting key.
+	 * @param string $default Default value.
+	 * @return string $setting_value Setting value or default value.
+	 */
+	public function get_option( $setting = '', $default = '' ) {
+		if ( empty( $this->gateway_settings ) ) {
+			return $default;
+		}
+
+		if ( ! empty( $this->gateway_settings[ $setting ] ) ) {
+			return $this->gateway_settings[ $setting ];
+		}
+
+		return $default;
+	}
+
+	/**
+	 * Whether the gateway and Payment Request Button (prerequisites for Apple Pay) are enabled.
+	 *
+	 * @return string Whether Apple Pay required settings are enabled.
+	 */
+	private function is_enabled() {
+		$gateway_enabled         = 'yes' === $this->get_option( 'enabled', 'no' );
+		$payment_request_enabled = 'yes' === $this->get_option( 'payment_request', 'yes' );
+
+		return $gateway_enabled && $payment_request_enabled;
+	}
+
+	/**
+	 * Trigger Apple Pay registration upon domain name change.
+	 */
+	public function verify_domain_on_domain_name_change() {
+		if ( $this->domain_name !== $this->get_option( 'apple_pay_verified_domain' ) ) {
+			$this->verify_domain_if_configured();
+		}
+	}
+
+	/**
+	 * Vefifies if hosted domain association file is up to date
+	 * with the file from the plugin directory.
+	 *
+	 * @return bool Wether file is up to date or not.
+	 */
+	private function verify_hosted_domain_association_file_is_up_to_date() {
```

Comment:
Let's change this function's name to `is_hosted_domain_association_file_up_to_date()`